stackit/internal/services/iaas/networkinterface/resource.go

```diff
@@ -553,19 +562,22 @@ func toCreatePayload(ctx context.Context, model *Model) (*iaas.CreateNICPayload,
 		}
 	}
 
-	allowedAddressesPayload := []iaas.AllowedAddressesInner{}
-
+	var allowedAddressesPayload *[]iaas.AllowedAddressesInner
 	if !(model.AllowedAddresses.IsNull() || model.AllowedAddresses.IsUnknown()) {
+		allowedAddresses := []iaas.AllowedAddressesInner{}
```

Comment:
Why do you need to instantiate the variable again?
You already declared it in line 564